2016-2017_Starstruck/CrossBot Competition/src/ServoLib.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * ServoLib.c
+ *
+ *  Created on: Dec 3, 2016
+ *      Author: Mikel Matticoli
+ */
+
+#include <main.h>
+
+/*
+ * This is the core for Mikel's Servo Module. This file should be placed
+ * in the src directory. Please note that this file will not compile
+ * if ServoLib.h has not been included properly.
+ */
+
+// Private function declarations
+
+/*
+ * Updates the speed of given motor based on current and target potentiometer values
+ */
+void servoLoop(ServoSystem servo);
+
+// Public function definitions
+ServoSystem servoInit(unsigned char potentiometerPort, unsigned char motorPort, bool motorInverted, int motorScale, int targetTolerance) {
+		printf("Initializing servo with potentiometer %d and motor %d\n\r", potentiometerPort, motorPort);
+		delay(100);
+		ServoSystem servo = {
+			.potentiometerPort = potentiometerPort,
+			.motorPort = motorPort,
+			.motorInverted = motorInverted,
+			.motorScale = motorScale,
+			.targetTolerance = targetTolerance,
+			.targetValue = malloc(sizeof(int *)),
+			.task = malloc(sizeof(TaskHandle *))
+		};
+		printf("Setting target");
```

Comment:
same thing for these prints, do we need them on the LCD or are they just for testing?